contracts/bonding/BondingCheckpoints.sol

```diff
@@ -0,0 +1,393 @@
+        if (_round > clock()) {
+            revert FutureLookup(_round, clock());
+        }
+
+        uint256 activeStake = totalActiveStakeCheckpoints[_round];
+
+        if (activeStake == 0) {
+            uint256 lastInitialized = checkedFindLowerBound(totalStakeCheckpointRounds, _round);
+
+            // Check that the round was in fact initialized so we don't return a 0 value accidentally.
+            if (lastInitialized != _round) {
+                revert MissingRoundCheckpoint(_round);
+            }
+        }
+
+        return activeStake;
+    }
+
+    /**
+     * @notice Gets the bonding state of an account at a given round.
+     * @dev In the case of delegators it is the amount they are delegating to a transcoder, while for transcoders this
+     * includes all the stake that has been delegated to them (including self-delegated).
+     * @param _account The account whose bonding state we want to get.
+     * @param _round The round for which we want to get the bonding state. Normally a proposal's vote start round.
+     * @return amount The active stake of the account at the given round including any accrued rewards. In case of
+     * transcoders this also includes all the amount delegated towards them by other delegators.
+     * @return delegateAddress The address the account delegated to. Will be equal to _account in case of transcoders.
+     */
+    function getBondingStateAt(address _account, uint256 _round)
+        public
+        view
+        virtual
+        returns (uint256 amount, address delegateAddress)
+    {
+        BondingCheckpoint storage bond = getBondingCheckpointAt(_account, _round);
+
+        delegateAddress = bond.delegateAddress;
+        bool isTranscoder = delegateAddress == _account;
+
+        if (bond.bondedAmount == 0) {
+            amount = 0;
+        } else if (isTranscoder) {
+            // Address is a registered transcoder so we use its delegated amount. This includes self and delegated stake
+            // as well as any accrued rewards, even unclaimed ones
+            amount = bond.delegatedAmount;
+        } else {
+            // Address is NOT a registered transcoder so we calculate its cumulative stake for the voting power
+            amount = delegatorCumulativeStakeAt(bond, _round);
+        }
+    }
+
+    /**
+     * @dev Gets the checkpointed bonding state of an account at a round. This works by looking for the last checkpoint
+     * at or before the given round and using the checkpoint of that round. If there hasn't been checkpoints since then
+     * it means that the state hasn't changed.
+     * @param _account The account whose bonding state we want to get.
+     * @param _round The round for which we want to get the bonding state.
+     * @return The {BondingCheckpoint} pointer to the checkpoints storage.
+     */
+    function getBondingCheckpointAt(address _account, uint256 _round)
+        internal
+        view
+        returns (BondingCheckpoint storage)
```

Comment:
Same here - I *think* this can be memory since this value isn't modified anywhere